core/system/src/unvic.c

```diff
@@ -179,6 +179,57 @@ void unvic_irq_disable(uint32_t irqn)
     }
 }
 
+/** Disables all interrupts for the currently active box.
+ *
+ * @internal
+ *
+ * This function keeps a state in the unprivileged vector table that will be
+ * used later on, in ::unvic_irq_enable_all to re-enabled previously disabled
+ * IRQs. */
+void unvic_irq_disable_all(void)
+{
+    int i;
+
+    /* Iterate over all the IRQs owned by the currently active box. */
+    for (i = 0; i < NVIC_VECTORS; i++) {
+        if (g_unvic_vector[i].id == g_active_box) {
+            /* Remember the state for this IRQ. The state is the NVIC one, so we
+             * are sure we don't enable spurious interrupts. */
+            g_unvic_vector[i].was_enabled = (bool) UNVIC_IS_IRQ_ENABLED(i);
+
+            /* Disable the IRQ. */
+            NVIC_DisableIRQ(i);
```

Comment:
This corner case might occur in the following cases:

* When the IRQ triggers while we are running `vIRQ_DisableAll()`.
* When we are running in an ISR for an IRQ and, even before calling `vIRQ_DisableAll()`, another IRQ with lower priority is triggered.

In those case the IRQ becomes pending. With the `vIRQ_DisableAll()` function we would disable that IRQ as well. From the ARMv7-M ARM:

> When an interrupt is disabled, interrupt assertion causes the interrupt to become pending, but the interrupt cannot become active. If an interrupt is active when it is disabled, it remains in the active state until this is cleared by a reset or an exception return. Clearing the enable bit prevents any new activation of the associated interrupt.

So in both cases, disabling a pending IRQ will leave it pending. It will probably be served as soon as someone calls `vIRQ_EnableAll()`.